<?xml version="1.0"?>
<launch> 
     <arg name = "horizontal_dispersion_x"             default = "0.22"    />
     <arg name = "horizontal_dispersion_y"             default = "0.0001"  />
     <arg name = "horizontal_dispersion_z"             default = "0.5"     />
     <arg name = "vertical_dispersion_x"               default = "0.2"     />    
     <arg name = "vertical_dispersion_y"               default = "0.0"     />    
     <arg name = "vertical_dispersion_z"               default = "0.0"     />   
     <arg name = "number_of_particles"                 default = "2000"    />    
     <arg name = "minimum_number_of_particles"         default = "1000"    /> 
     <arg name = "measurement_noise"                   default = "1.0"     />      
     <arg name = "process_noise"                       default = "1.0"     />         
    
     <node pkg="nrg_gas_source_localization"  type="nrg_gas_source_localization_node" name="nrg_gas_source_localization_node" output="screen">

          <!--NRGGas-->
          <param name = "horizontal_dispersion_x"      value = "$(arg horizontal_dispersion_x)"     />  
          <param name = "horizontal_dispersion_y"      value = "$(arg horizontal_dispersion_y)"     />  
          <param name = "horizontal_dispersion_z"      value = "$(arg horizontal_dispersion_z)"     />  
          <param name = "vertical_dispersion_x"        value = "$(arg vertical_dispersion_x)"       /> 
          <param name = "vertical_dispersion_y"        value = "$(arg vertical_dispersion_y)"       />  
          <param name = "vertical_dispersion_z"        value = "$(arg vertical_dispersion_z)"       />  

          <!--NRGGasSourceLocalization-->
          <rosparam param = "state_space_bounds_x">    [-100.0, 100.0]     </rosparam>
          <rosparam param = "state_space_bounds_y">    [-100.0, 100.0]     </rosparam>
          <rosparam param = "state_space_bounds_z">    [0.0, 25.0]        </rosparam>
          <rosparam param = "state_space_bounds_rate"> [0.0, 20000.0]      </rosparam>
          <param name = "number_of_particles"          value = "$(arg number_of_particles)"         />  
          <param name = "minimum_number_of_particles"  value = "$(arg minimum_number_of_particles)" /> 
          <param name = "measurement_noise"            value = "$(arg measurement_noise)"           />  
          <param name = "process_noise"                value = "$(arg process_noise)"               />  

     </node>

</launch>
